build/purger/purger.py

```diff
@@ -263,17 +220,30 @@ def update_user_activity(
     return network_to_due_users
 
 
+def get_discovery_room(api: GMatrixHttpApi, network_value: int) -> Optional[RoomInfo]:
+
+    server = urlparse(api.base_url).netloc
+    discovery_room_alias = make_room_alias(ChainID(network_value), DISCOVERY_DEFAULT_ROOM)
+    local_room_alias = f"#{discovery_room_alias}:{server}"
+
+    try:
+        room_id = api.get_room_id(local_room_alias)
+        return RoomInfo(room_id, discovery_room_alias, server)
+    except MatrixError as ex:
+        click.secho(f"Could not find room {discovery_room_alias} with error {ex}")
+
+    return None
+
+
 def _fetch_new_members_for_network(
     api: GMatrixHttpApi, user_activity: Dict[str, int], discovery_room: RoomInfo, current_time: int
 ) -> None:
     try:
-        response = api.get_room_members(discovery_room.room_id)
+        response = api._send("GET", f"/_synapse/admin/v1/rooms/{discovery_room.room_id}/members")
         room_members = [
-            event["state_key"]
-            for event in response["chunk"]
-            if event["content"]["membership"] == "join"
-            and event["state_key"].split(":")[1] == urlparse(api.base_url).netloc
-            and not event["state_key"].find("admin")
+            member
+            for member in response["members"]
+            if member.split(":")[1] == urlparse(api.base_url).netloc and not member.find("admin")
```

Comment:
This will unnecessarily re-run `urlparse` for every user in the list.

Also `<str>.find()` isn't correct here (since it will only be falsey if the string starts with the search string, which will not be the case here).
Either `"admin" in member` or `member.startswith("@admin")` would work.